src/liquid_parquet/src/cache/store.rs

```diff
@@ -234,7 +275,54 @@ impl CacheStore {
         self.budget.add_used_disk_bytes(bytes.len());
     }
 
-    pub(super) fn insert(&self, entry_id: CacheEntryID, mut batch_to_cache: CachedBatch) {
+    fn write_to_disk_async(&self, entry_id: &CacheEntryID, liquid_array: &LiquidArrayRef) {
```

Comment:
I think for write_to_disk_async, this function itself should be async, so that the .await call will yield to the scheduler etc.

The block_in_place only moves away other tasks on this thead